addons/account/tests/test_chart_template.py

```diff
@@ -0,0 +1,253 @@
+                    'repartition_type': 'tax',
+                }),
+            ],
+        })
+        cls.env['ir.model.data'].create({
+            'name': 'test_tax_1_template',
+            'module': 'account',
+            'res_id': cls.tax_1_template.id,
+            'model': 'account.tax.template',
+        })
+        cls.tax_2_template = cls.env['account.tax.template'].create({
+            'name': 'Tax 2',
+            'chart_template_id': cls.chart_template.id,
+            'invoice_repartition_line_ids': [
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'base',
+                }),
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                }),
+            ],
+            'refund_repartition_line_ids': [
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'base',
+                }),
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                }),
+            ],
+        })
+        cls.env['ir.model.data'].create({
+            'name': 'test_tax_2_template',
+            'module': 'account',
+            'res_id': cls.tax_2_template.id,
+            'model': 'account.tax.template',
+        })
+
+        cls.fiscal_position_template = cls.env['account.fiscal.position.template'].create({
+            'name': 'Fiscal Position',
+            'chart_template_id': cls.chart_template.id,
+            'country_id': cls.env.ref('base.be').id,
+            'auto_apply': True,
+        })
+        cls.env['ir.model.data'].create({
+            'name': 'fiscal_position_template',
+            'module': 'account',
+            'res_id': cls.fiscal_position_template.id,
+            'model': 'account.fiscal.position.template',
+        })
+
+        fiscal_position_tax_template_1 = cls.env['account.fiscal.position.tax.template'].create({
+            'tax_src_id': cls.tax_1_template.id,
+            'tax_dest_id': cls.tax_2_template.id,
+            'position_id': cls.fiscal_position_template.id,
+        })
+        cls.env['ir.model.data'].create({
+            'name': 'fiscal_position_tax_template_1',
+            'module': 'account',
+            'res_id': fiscal_position_tax_template_1.id,
+            'model': 'account.fiscal.position.tax.template',
+        })
+
+        cls.company_1 = cls.env['res.company'].create({
+            'name': 'TestCompany1',
+            'country_id': cls.env.ref('base.be').id,
+        })
+        cls.env.company = cls.company_1
+        cls.chart_template.try_loading(company=cls.company_1)
+
+        cls.partner_1 = cls.env['res.partner'].create({
+            'name': 'Partner 1',
+            'country_id': cls.env.ref('base.be').id,
+        })
+
+    def test_update_taxes_from_templates(self):
```

Comment:
What happens if you delete a tax from the templates or from the fiscal position mappings?

Reply:
Added a test that enforces that if a tax was deleted, it will be recreated (that was the case before this commit) and that only fiscal position mappings concerning taxes that did not exist before updating (deleted ones or new ones) will be created